GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh

```diff
@@ -26,10 +26,36 @@ echo "%MSG-MG5 random seed used for the run = $rnum"
 ncpu=${4}
 echo "%MSG-MG5 thread count requested = $ncpu"
 
-echo "%MSG-MG5 residual arguments = ${@:5}"
+echo "%MSG-MG5 residual/optional arguments = ${@:5}"
+
+if [ -n "${5}" ]; then
+  use_gridpack_env=${5}
+  echo "%MSG-MG5 use_gridpack_env = $use_gridpack_env"
+fi
+
+if [ -n "${6}" ]; then
+  scram_arch_version=${6}
+  echo "%MSG-MG5 override scram_arch_version = $scram_arch_version"
+fi
+
+if [ -n "${7}" ]; then
+  cmssw_version=${7}
+  echo "%MSG-MG5 override cmssw_version = $cmssw_version"
+fi
 
 LHEWORKDIR=`pwd`
 
+if [ "$use_gridpack_env" = false -a -n "$scram_arch_version" -a -n  "$cmssw_version" ]; then
+  echo "%MSG-MG5 CMSSW version = $cmssw_version"
+  export VO_CMS_SW_DIR=/cvmfs/cms.cern.ch
+  source $VO_CMS_SW_DIR/cmsset_default.sh
```

Comment:
hi @bendavid - i'm surprised that these two lines would be needed? the job is running within a cms environment after all

Reply:
This is handling the case where a site has configured a non-cvmfs source for the release area (where there would presumably be weaker guarantees on which releases are available)